Remove empty trailing spaces in markdown files

Dorian opened the pull request:
They have no useful purpose (and by default my vim config removes them :) )

One-liner used:

```
Dir['**/*.md'].each { |f| File.write(f, File.read(f).lines.map(&:rstrip).join("\n") + "\n") }
```

(I tried configuring [`markdownlint`](https://github.com/mivok/markdownlint) but it didn't work, would be nice to have a markdown linter as part of the CI)

bbatsov commented:
Those are auto-generated, so I guess there's actually a bug in the code generating them.

Dorian commented:
@bbatsov Of course, thanks :)

Dorian commented:
@bbatsov Also, because why not :D, I started fixing `markdownlint` offenses (I'm down from 449 to 26 offenses with `system("mdl -r #{`mdl -l | egrep -v "(MD002|MD013|MD024)"`.lines[1..-1].map { |l| l.split.first }.join(',')} **/*.md")`).

I guess a lot of them will have to be merged in some way inside the rake task (but anyway, it's easier to have a target for the generation later).

Ok, so what do you think? Is having a markdown linter too much? Linting inside a linter is too meta? 😄 

Most common offenses are line length and headers with same content but I disabled them for now:

```ruby
>> puts `mdl -r #{`mdl -l`.lines[1..-1].map { |l| l.split.first }.join(',')} **/*.md`.lines.map { |l| l.split[1] }.group_by(&:itself).map { |a, b| [a, b.size] }.sort_by(&:last).reverse.map { |l| "- " + l.join(": ") }.join("\n")
```

```
# master
- MD013: 1188
- MD024: 530
- MD031: 214
- MD012: 155
- MD002: 54
- MD009: 18
- MD004: 14
- MD005: 12
- MD014: 11
- MD034: 5
- MD032: 5
- MD036: 3
- MD006: 3
- MD033: 3
- MD029: 2
- MD038: 1
- MD023: 1
- MD027: 1
- MD007: 1

# markdown-linter branch

- MD013: 1187
- MD024: 530
- MD002: 54
- MD014: 11
- MD034: 5
- MD036: 3
- MD033: 3
- MD029: 2
- MD038: 1
- MD023: 1
```

([RULES.md](https://github.com/mivok/markdownlint/blob/master/docs/RULES.md))

bbatsov commented on manual/cops_metrics.md:
```diff
@@ -38,7 +38,7 @@ Attribute | Value
 Max | 25
-ExcludedMethods | 
+ExcludedMethods |
 
```
Probably you should also change whatever is adding 2 blank lines here instead of one.

bbatsov commented on tasks/cops_documentation.rake:
```diff
@@ -32,7 +32,7 @@ task generate_cops_documentation: :yard do
     default_status = config.cop_enabled?(cop) ? 'Enabled' : 'Disabled'
     supports_autocorrect = cop.new.support_autocorrect? ? 'Yes' : 'No'
-    content << "#{default_status} | #{supports_autocorrect}"
+    content << "#{default_status} | #{supports_autocorrect}".strip
```
I don't see the point for doing `strip` here. Where would an extra space come from?

bbatsov commented on tasks/cops_documentation.rake:
```diff
@@ -64,7 +64,7 @@ task generate_cops_documentation: :yard do
     content << "--- | ---\n"
     pars.each do |par|
-      content << "#{par.first} | #{format_table_value(par.last)}\n"
+      content << "#{par.first} | #{format_table_value(par.last)}".strip + "\n"
```
This also looks a bit strange to me.

Dorian commented:
@bbatsov Looking good :)
